docs/source/llms/langchain/autologging.rst

```diff
@@ -0,0 +1,262 @@
+
+.. code-block::
+
+    import mlflow
+
+    mlflow.langchain.autolog()
+
+    # Enable other optional logging
+    # mlflow.langchain.autolog(log_models=True, log_datasets=True)
+
+    # Your LangChain model code here
+    ...
+
+Once you have invoked the chain, you can view the logged traces and artifacts in the MLflow UI.
+
+.. figure:: ../../_static/images/llms/tracing/langchain-tracing.gif
+    :alt: LangChain Tracing via autolog
+    :width: 100%
+    :align: center
+
+
+Configure Autologging
+---------------------
+
+MLflow LangChain autologging can log various information about the model and its inference. **By default, only trace logging is enabled**, but you can enable autologging of other information by setting the corresponding parameters when calling :py:func:`mlflow.langchain.autolog()`. For other configurations, please refer to the API documentation.
+
+.. list-table::
+    :widths: 20 20 30 30
+    :header-rows: 1
+
+    * - Target
+      - Default
+      - Parameter
+      - Description
+    * - Traces
+      - ``true``
+      - ``log_traces``
+      - Whether to generate and log traces for the model. See `MLflow Tracing <../tracing/index.html>`_ for more details about tracing feature.
+    * - Model Artifacts
+      - ``false``
+      - ``log_models``
+      - If set to ``True``, the LangChain model will be logged when it is invoked. Supported models are `Chain`, `AgentExecutor`, `BaseRetriever`, `RunnableSequence`, `RunnableParallel`, `RunnableBranch`, `SimpleChatModel`, and `ChatPromptTemplate` (Additional model types will be supported in the future.)
+    * - Model Signatures
+      - ``false``
+      - ``log_model_signatures``
+      - If set to ``True``, :py:class:`ModelSignatures <mlflow.models.ModelSignature>` describing model inputs and outputs are collected and logged along with Langchain model artifacts during inference.
+    * - Input Example
+      - ``false``
+      - ``log_input_examples``
+      - If set to ``True``, input examples from inference data are collected and logged along with LangChain model artifacts during inference.
+    * - Inputs and Outputs (**Deprecated**)
+      - ``false``
+      - ``log_inputs_outputs``
+      - If set to ``True``, the inputs and outputs will be logged when the model is invoked. This feature is deprecated and will be removed in the future. Please use ``log_traces`` instead.
+
+For example, to disable logging of traces, and instead enable model logging, run the following code:
+
+.. code-block::
+
+    import mlflow
+
+    mlflow.langchain.autolog(
+        log_traces=False,
+        log_models=True,
+    )
+
+.. note::
+
+    When you enable any optional logging other than tracing, MLflow creates an MLflow Run in the active experiment and logs artifacts there. Each inference call logs those artifacts into a separate directory named `artifacts-<session_id>-<idx>`, where `session_id` is a randomly generated uuid, and `idx` is the index of the inference call.
```

Comment:
This seems apply to old callback. Can we remove this part here since log_inputs_outputs is also deprecated?